### ISSUE TITLE
Disappearing Stations

### ISSUE THREAD

**Issue by reillysiemens** [PromyLOPh/pianobar#693]
### Disappearing Stations
As of today I noticed that whenever I select a station with pianobar that station is lost from the station list thereafter and does not re-appear unless I add it again. Though, that re-adding is **not** permanent and after closing the program the station disappears. The stations do not appear to _actually_ be gone from Pandora. They are visible in their mobile application at least.

### Your environment
* pianobar `2019.02.14` and `2019.02.14-dev` (`master`)
* Fedora 31 (kernel `5.5.7`)
* ffmpeg 4.2.2
* My config file:
   ```
   user = ██████████████
   password_command = /usr/bin/gpg2 --decrypt ~/.config/pianobar/password

   format_nowplaying_song = ^[[32m%t^[[0m by ^[[34m%a^[[0m on %l^[[31m%r^[[0m%@%s
   ban_icon =  ⛔
   love_icon =  ❤️
   tired_icon =  💤
   format_nowplaying_station = Station ^[[35m%n^[[0m
   format_list_song = %i) %a - %t%r (%d)%@%s
   ```
   Note that the `^[` in the above configuration are `ESC [`, the ANSI Control Sequence Introducer.

### Steps to reproduce
Start pianobar, select a station, close pianobar. The selected station disappears from the station list the next time pianobar is started.

### Expected behaviour
Start pianobar, select a station, close pianobar. The selected station remains in the station list the next time pianobar is started.

### Actual behaviour
Start pianobar, select a station, close pianobar. The selected station disappears from the station list the next time pianobar is started.

When I first encountered this I saw an error message that said
```
/!\ Error: Autostart station not found.
```
but after that I don't there there was a station to auto-start from anymore, so I don't see that error now.

**Comment by PromyLOPh** [comment]
I cannot reproduce this issue here. Must be some server-side glitch. Is
it still happening? Can you reproduce it for any of your stations or
just a specific one?

**Comment by reillysiemens** [comment]
It's definitely a bizarre issue! Since I hadn't changed anything about my pianobar usage for a while I suspected it might be a server-side thing.

I can reproduce it for the 4 that I lost yesterday:
- `"Frankey & Sandrino"`
- `"French Traditional Radio"`
- `"RJD2"`
- `"Wax Taylor"`

I can re-add all of those when I start up pianobar and listen to music from them, but when I close and re-open the application I get another
```
/!\ Error: Autostart station not found.
```
and the station is missing.

I also lost a
- `"Yoshida Brothers"`

station today while attempting to reproduce it.

**Comment by PromyLOPh** [comment]
Could you please compile the debug branch I just pushed with

	make CFLAGS="-O2 -g"

and then run

	PIANOBAR_DEBUG=1 ./pianobar 2>> stderr.txt

twice (so we can see the server’s response before and after the glitch).
It should print all network traffic to stderr.txt. You can either have a
look yourself and post relevant findings or – if you trust me – send a
redacted version (remove username, password and userAuthToken) to
lars+pianobar@6xq.net. My GPG key is here: https://6xq.net/08D8092A.gpg

**Comment by cithraul** [comment]
I've encountered the same issue. My autostart station suddenly disappeared from the list. It is still listed on my pandora account though. No changes made to pianobar config.
I can add the station back via pianobar, but it disappears upon the next pianobar startup.
My system is : 
Ubuntu 18.04.1 LTS 
kernel 4.15.0-54-generic
ffmpeg version 3.4.2-2
Pianobar version (2019.02.14-dev)

**Comment by Carter90** [comment]
Same issue 
Ubuntu 16.04 4.15.0-72-generic pianobar 2015.11.22
The autostart_station vanishes even when setting a new one it vanishes too. 

config is very basic 
```
user = ###########
password = ##########
autostart_station = ###########
```

**Comment by dcode** [comment]
I'm encountering this on Mac OS X too with build `2018.06.22` installed from homebrew. But it's not just autoplay stations. Namely, I had "Cake Radio" (station ID: 4441620509232913515) and it was an autostart station, but now it's not in the list. I can re-add it and play during that session. The station does show up in the web UI still too (shows created 3/4/2020). 

When I first experienced this, I noticed the station list in Pianobar was skipping numbers, and after messing with it (started deleting stations to see if that helped, but Cake never came back) I can see 95 stations. When I use the web player, I can count the elements of the parent div (w/ class name of "MyStationsList__listContainer") and there are 230 stations listed (don't judge, this account was hooked up to Alexa for a bit and my kids went nuts).

**Comment by PromyLOPh** [comment]
Thanks @dcode, I’m able to reproduce it now. Can someone please confirm
the change I pushed to the branch fix693 fixes the issues? Is works for
me.

**Comment by dcode** [comment]
YES! Adding the following to the homebrew recipe above the install method and reinstalling from source fixed the problem.

```ruby
  patch do
      url "https://github.com/PromyLOPh/pianobar/commit/e07f023cb5cdbb0c24d6dccd9f3b771078831dcb.patch"
      sha256 "4ceccafca1576d804686f6f1ba8f2141ef1ab0e8c28493560922b06cbcdabf91"
  end
```

```bash
brew reinstall pianobar -s
```

Now, I need to clean up my station list that was hiding. 😝

**Comment by PromyLOPh** [comment]
Excellent. I merged the commit into master and will prepare a new
release in the coming days.

**Comment by bryan00a** [comment]
I was also having this problem, works great again! Thank you!

**Comment by cithraul** [comment]
Thanks for the quick turnaround PromyLOPh

**Comment by PromyLOPh** [comment]
I just released version 2020.04.05.

**Comment by reillysiemens** [comment]
Sorry I wasn't able to get you those debugging logs in time. :sweat_smile:

Thanks for finding a solution so quickly, @PromyLOPh! :tada:
